### ISSUE TITLE
Should with custom error fails

### ISSUE THREAD

**Issue by yanickrochon** [tj/should.js#195]
I have this custom error defined :

``` javascript
var AsyncTimeoutError = function AsyncTimeoutError(msg) {
  msg && (this.message = msg);
  Error.apply(this, arguments);
  Error.captureStackTrace(this, AsyncTimeoutError);
};
require('util').inherits(AsyncTimeoutError, Error);
AsyncTimeoutError.prototype.name = AsyncTimeoutError.name;
```

Manual testing gives

```
var err = new AsyncTimeoutError('foo');

console.log(err instanceof Error);
// -> true
```

However, should fails

``` javascript
var err = new AsyncTimeoutError('foo');

err.should.be.an.Error;
```

**Comment by btd** [comment]
Yes, i used `util.isError` (and it return false for your error), that check Object.prototype.toString on this object. But yes i agree that it is bug.

**Comment by btd** [comment]
Fixed in 3.1.4

**Comment by yanickrochon** [comment]
Cheers!
